### DISCUSSION TITLE
A differentiation question

### DISCUSSION THREAD

**Discussion by vgivanovic** [zellij-org/zellij#738] [General]
Greenspun's Tenth Rule says, "Any sufficiently complicated C or Fortran program contains an ad hoc, informally-specified, bug-ridden, slow implementation of half of Common Lisp." (Greenspun's rule has been extended to use the X Window system or TCP/IP instead of Common Lisp.)

In this vein, why isn't Zellij just a partial re-implementation of GNU Emacs? Please understand that I'm not trying to start a war, or to denigrate anyone's preferences. I'm asking a high-level, general question of differentiation, "How is Zellij different/better than Emacs?" (Emacs, tongue firmly in cheek, has been described as an operating system disguised as an editor.)

At first glance, I don't see that Zellij does anything that Emacs doesn't do. I'm prepared to believe that Zellij is much faster than Emacs, but Emacs is fast enough. I'm not a vi user, but I suspect that neovim, for example, does much of what Zellij does and is fast enough. So, let me rephrase, "How is Zellij different/better than Emacs or Neovim?" All three have an extension system (WASM, Elisp, and Lua respectively) that hooks into their core, so in principle, all three are equivalent. It's just a matter of programming ;-) I suppose one should include Atom in this list of possible equivalents. 

I didn't see an obvious RATIONALE.md, so I'm suggestion that one be created that answers questions like, "Why Zellij?",  "Why Rust?", or "Why did we design Zellij the way we did?" For example, you state that your plugin system uses WASM because many programming languages compile to WASM, making creating plugins easier. Great! That reasoning belongs in RATIONALE.md.

**Comment by qballer** [comment]
hi @vgivanovic, zellij is meant to be a workspace which primary handles terminals and custom plugins within a terminal emulator. While emacs/vim and even atom have some windowing systems they are meant first as text editors (may the emacs folks forgive me). This question can be better asked by comparing zellij to tmux which is much closer to it - yet has a different approach to things.
As for why rust - well the initial project creator rightfully thought zellij deserves some system programing language and Rust seemed most appealing. Zellij can be written in C, CPP and maybe even golang IMO but it wasn't. 

Have you tried using zellij ? Do you like it?
